src/vstest.console/Processors/ListExtensionsArgumentProcessor.cs

```diff
@@ -0,0 +1,236 @@
+
+    internal class ListExtensionsArgumentProcessorCapabilities : BaseArgumentProcessorCapabilities
+    {
+        private string commandName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ListExtensionsArgumentProcessorCapabilities"/> class.
+        /// </summary>
+        /// <param name="commandName">Name of the command</param>
+        public ListExtensionsArgumentProcessorCapabilities(string commandName)
+        {
+            this.commandName = commandName;
+        }
+
+        public override string CommandName => this.commandName;
+
+        /// <inheritdoc />
+        public override bool AllowMultiple => false;
+
+        /// <inheritdoc />
+        public override bool IsAction => true;
+    }
+
+    internal abstract class ListExtensionsArgumentProcessor : IArgumentProcessor
+    {
+        private Lazy<IArgumentProcessorCapabilities> metadata;
+        private Lazy<IArgumentExecutor> executor;
+        private Func<IArgumentExecutor> getExecutor;
+        private Func<IArgumentProcessorCapabilities> getCapabilities;
+
+        public ListExtensionsArgumentProcessor(Func<IArgumentExecutor> getExecutor, Func<IArgumentProcessorCapabilities> getCapabilities)
+        {
+            this.getExecutor = getExecutor;
+            this.getCapabilities = getCapabilities;
+        }
+
+        public Lazy<IArgumentExecutor> Executor
+        {
+            get
+            {
+                if (this.executor == null)
+                {
+                    this.executor = new Lazy<IArgumentExecutor>(getExecutor);
+                }
+
+                return this.executor;
+            }
+
+            set
+            {
+                this.executor = value;
+            }
+        }
+
+        public Lazy<IArgumentProcessorCapabilities> Metadata
+        {
+            get
+            {
+                if (this.metadata == null)
+                {
+                    this.metadata = new Lazy<IArgumentProcessorCapabilities>(getCapabilities);
+                }
+
+                return this.metadata;
+            }
+        }
+    }
+
+    #region List discoverers
+    /// <summary>
+    /// Argument Executor for the "/ListDiscoverers" command line argument.
+    /// </summary>
+    internal class ListDiscoverersArgumentProcessor : ListExtensionsArgumentProcessor
+    {
+        private const string CommandName = "/ListDiscoverers";
+
+        public ListDiscoverersArgumentProcessor()
+            : base(() => new ListDiscoverersArgumentExecutor(), () => new ListExtensionsArgumentProcessorCapabilities(CommandName))
+        {
+        }
+    }
+
+    internal class ListDiscoverersArgumentExecutor : IArgumentExecutor
+    {
+        public void Initialize(string argument)
+        {
+        }
+
+        public ArgumentProcessorResult Execute()
+        {
+            ConsoleOutput.Instance.WriteLine(CommandLineResources.AvailableDiscoverersHeaderMessage, OutputLevel.Information);
```

Comment:
How come `AvailableDiscoverersHeaderMessage` resources strings already exists? Are we having any more unused resources strings?

Reply:
I think these were copied as-is from older version of test platform.